todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.kt

```diff
@@ -61,18 +60,17 @@ class AddEditTaskFragment : Fragment() {
     }
 
     private fun setupFab() {
-        activity.findViewById<FloatingActionButton>(R.id.fab_edit_task_done).apply {
-            setImageResource(R.drawable.ic_done)
-            setOnClickListener { viewDataBinding.viewmodel?.saveTask() }
-        }
+        activity?.findViewById<FloatingActionButton>(R.id.fab_edit_task_done)?.apply {
+                setImageResource(R.drawable.ic_done)
+                setOnClickListener { viewDataBinding.viewmodel?.saveTask() }
+            }
     }
 
     private fun setupActionBar() {
         (activity as AppCompatActivity).supportActionBar?.setTitle(
-                if (arguments != null && arguments.get(ARGUMENT_EDIT_TASK_ID) != null)
-                    R.string.edit_task
-                else
-                    R.string.add_task
+                arguments?.get(ARGUMENT_EDIT_TASK_ID)
```

Comment:
You don't need the qualified name for kotlin.run (just use `run {...}`). Also, have you tested this? The line after `?:` will only trigger if the last line of the .run { ... } block is null, which it never is in this case. if you switch it to apply {...}, it will give you what you want.

I find using standard functions combined with `?:` is prone to mistakes such as this, so here I prefer traditional `if` statements. Do you mind reverting to the original but maybe assigning `arguments` to a temporary variable `args` so you can null check it atomically?

Reply:
It should be really wrong that the run con-cats with next run, my bad coding-style here. 
Well how about this change instead if-statement
```
arguments?.get(ARGUMENT_EDIT_TASK_ID)
                        ?.let { R.string.edit_task }
                        ?: run { R.string.add_task }
```
I prefere to use ```?.``` operator to instead more than one step check in codes, I think the Kotlin introduce this operator aiming to avoide this c-style.